frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -127,28 +129,45 @@ export const billingLogic = kea<billingLogicType>([
             },
         ],
     }),
-    loaders(({ actions }) => ({
+    loaders(({ actions, values }) => ({
         billing: [
             null as BillingV2Type | null,
             {
                 loadBilling: async () => {
+                    actions.resetBillingError()
                     const response = await api.get('api/billing-v2')
 
                     return parseBillingResponse(response)
                 },
 
                 updateBillingLimits: async (limits: { [key: string]: string | null }) => {
+                    actions.resetBillingError()
                     const response = await api.update('api/billing-v2', { custom_limits_usd: limits })
 
                     lemonToast.success('Billing limits updated')
                     return parseBillingResponse(response)
                 },
 
                 deactivateProduct: async (key: string) => {
-                    const response = await api.get('api/billing-v2/deactivate?products=' + key)
-                    lemonToast.success('Product unsubscribed')
-                    actions.reportProductUnsubscribed(key)
-                    return parseBillingResponse(response)
+                    actions.resetBillingError()
+                    try {
+                        const response = await api.getResponse('api/billing-v2/deactivate?products=' + key)
+                        const jsonRes = await getJSONOrNull(response)
+                        lemonToast.success('Product unsubscribed')
+                        actions.reportProductUnsubscribed(key)
+                        return parseBillingResponse(jsonRes)
+                    } catch (error: any) {
+                        lemonToast.error(`${error.detail}`, {
```

Comment:
What happens if there is another error throw? The "See Invoices" doesn't really make sense, right?

Reply:
Great point, thanks for reminding me. This is a bit messy but I think I'll check against the error detail to see if it's about invoices and only display the link to the stripe page if it is.